ci/scripts/towncrier_automation.py

```diff
@@ -0,0 +1,89 @@
+#!python3
+from __future__ import annotations
+
+import argparse
+import subprocess
+from typing import TYPE_CHECKING
+
+from packaging import version
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+
+class Args(argparse.Namespace):
+    version: str
+    dry_run: bool
+
+
+def parse_args(argv: Sequence[str] | None = None) -> Args:
+    parser = argparse.ArgumentParser(
+        prog="towncrier-automation",
+        description=(
```

Comment:
I removed `usage` here, it’s autogenerated, and can be accessed via `-h`/`--help`